File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.py
def solution(x):
    # x가 각 자리수의 합으로 나누어 떨어지는 지를 판별
    # 각 자리수의 합 구하기
    sum = 0
    for digit in list(map(int, str(x))):
        sum += digit
    # 합으로 나누어 떨어지는 지 여부를 판단하기
    return (x % sum) == 0

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        int[] chars = String.valueOf(x).chars().map(c -> Character.getNumericValue(c)).toArray();
        int sum = 0;
        for(int num: chars){
            System.out.println(num);   
            sum += num;
        }
        return x % sum == 0;
    }
}